src/category_theory/concrete_category/bundled_hom.lean

```diff
@@ -57,7 +58,8 @@ intros; apply 𝒞.hom_ext;
   simp only [𝒞.id_to_fun, 𝒞.comp_to_fun, function.left_id, function.right_id]
 
 /-- A category given by `bundled_hom` is a concrete category. -/
-instance concrete_category : concrete_category (bundled c) :=
+-- The linter `dangerous_instance` fails, because it generates the type-class problem bundled_hom ?m
```

Comment:
Again, I can't tell if this is truly dangerous or not without knowing the library better. My gut feeling is that if we sometimes have to ignore the linter "because here it's okay," the linter test probably isn't right. But if here it's okay just by circumstance, maybe we should trust the linter and this shouldn't be an instance?

Reply:
In this case it is okay because there are 0 instances for the `bundled_hom` class, so it will only check local constants. 

It will be problematic if there are any `bundled_hom`-instances that are recursive, i.e. that have type-class arguments themselves (in a way that can result in an infinite loop).

I guess I could test for this, but it sounds pretty tricky.

Reply:
Hmm. There are a few instances of `bundled_hom` (https://leanprover-community.github.io/mathlib_docs/category_theory/concrete_category/bundled_hom.html#category_theory.bundled_hom), but nothing that will make it loop.

Maybe we should move the warning into the docstring to make it more visible (here and above)? These are potentially bad instances. If the "wrong" good instances get added to the library, these could cause issues. It seems worth it to flag that as clearly as possible.

Reply:
A possible solution is to include `hom` type in the `bundled` signature.

Reply:
> There are a few instances of `bundled_hom`

I might have accidentally checked in the file `bundled_hom` instead of a file that imports all of mathlib, but I don't see any instances of the class `bundled_hom` in the file you linked.

I'm happy to add the warning to the docstring. These instances could indeed become dangerous if we do seemingly-innocuous things.

Reply:
Oh, in the file `unbundled_hom` there is an instance for `bundled_hom`.

Reply:
Oh! It lists all instances just below the class. Nice!

Reply:
@urkud: adding `hom` as a field instead of an argument would definitely solve these false positives, and that seems like a good thing to do. I don't know whether it is a problem that `bundled_hom.hom` is not syntactically equal to whatever it is defined as.

Reply:
@fpvandoorn I meant adding `hom` as an unused argument to the `bundled` function.

Reply:
Oh, sorry, you said adding it as an argument to `bundled` (I misread it as `bundled_hom`). Doesn't that run into issues if you need to synthesize `category (bundled c ?m)`? I don't know, maybe that works.